src/molecules/BottomSheet/BottomSheet.native.js

```diff
@@ -0,0 +1,190 @@
+    };
+  },
+  childrenStyle: ({ theme }) => {
+    return {
+      backgroundColor: theme.colors.background[200],
+    };
+  },
+  linearGradient: () => {
+    return {
+      transform: [
+        {
+          scaleX: 1,
+        },
+        { scaleY: -1 },
+      ],
+    };
+  },
+};
+
+const HeaderContainer = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[200]};
+  border-top-right-radius: 8px;
+  border-top-left-radius: 8px;
+  box-shadow: ${(props) => `0px -4px 15px ${props.theme.colors.primary[930]}`};
+  elevation: 4;
+`;
+
+const BottomSheetDragBar = styled(View)`
+  background-color: ${(props) => props.theme.colors.shade[920]};
+  border-radius: 4px;
+`;
+
+const BottomSheet = forwardRef(
+  (
+    {
+      snapPoint = DEFAULT_SNAP_POINT,
+      children,
+      onBackDropPress = () => {},
+      onBackButtonPress = () => {},
+      onOpened = () => {},
+      onClosed = () => {},
+      adjustToContentHeight = false,
+      alwaysOpen = 0,
+    },
+    ref,
+  ) => {
+    const theme = useTheme();
+    const [headerHeight, setHeaderHeight] = useState(0);
+    const [contentHeight, setContentHeight] = useState(0);
+    const bottomsheetChildrenGroupByType = reactChildrenGroupByType(children);
+
+    const headerComponent = bottomsheetChildrenGroupByType[Header];
+    const footerComponent = bottomsheetChildrenGroupByType[Footer];
+    const contentComponent = bottomsheetChildrenGroupByType[Content];
+
+    if (headerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Header\` but found ${headerComponent.length}`,
+      );
+    }
+
+    if (footerComponent?.length > 1) {
+      throw new Error(
+        `expected to have single \`BottomSheet.Footer\` but found ${footerComponent.length}`,
+      );
+    }
+
+    const handleHeaderLayoutChange = useCallback((e) => {
+      setHeaderHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    const handleContentLayoutChange = useCallback((e) => {
+      setContentHeight(e.nativeEvent.layout.height);
+    }, []);
+
+    let contentContainerHeight = DEFAULT_SNAP_POINT - headerHeight;
+    if (alwaysOpen > 0) {
+      contentContainerHeight = alwaysOpen - headerHeight;
+    }
+    const isScrollableContent = contentHeight > contentContainerHeight;
+
+    return (
+      <RNModalize
+        ref={ref}
+        snapPoint={snapPoint}
+        HeaderComponent={
+          <View onLayout={handleHeaderLayoutChange}>
+            <HeaderContainer>
+              <Flex alignItems="center">
+                <Space padding={[1, 0, 1.5, 0]}>
+                  <View>
+                    <Size height={0.5} width={8}>
+                      <BottomSheetDragBar />
+                    </Size>
+                  </View>
+                </Space>
+              </Flex>
+              {headerComponent?.length ? (
+                <>
+                  {headerComponent}
+                  <Divider color="shade.920" horizontal />
```

Comment:
Doesn't `Header` has a default padding surrounding? If it won't have we'll end up being creating inconsistent UIs everytime or everytime someone will add different paddings. The goal of Header, Footer and content was to have surrounding fixed paddings and then people can do whatever they want inside them

Reply:
We need to confirm this with design. At least I did not see any specific padding for these components.

Reply:
Confirmed with design, paddings are there only for content, but not for Header and Footer. https://razorpay.slack.com/archives/CL74S0A9Z/p1594901071043100?thread_ts=1594807904.029700&cid=CL74S0A9Z